### ISSUE TITLE
Junction restriction manager is not using the cache it has

### ISSUE THREAD

**Issue by kianzarrin** [CitiesSkylinesMods/TMPE#902]
Junction restriction manager calculates and stores default traffic rules on segment end update but `IsDefault<TrafficRule>Allowed` is not using the cached value most of the times. This means they are recalculated for every path find.

We need to introduce new internal methods:
`CalcualteDefault<TrafficRule>Allowed()` : called on segment end update
`Calcualte<TrafficRule>IsConfigurable()`  : called on segment end update

This way `IsDefault<TrafficRule>Allowed()` `Is<TrafficRule>IsConfigurable()` need only access the cached value which is fast.

I will elaborate more on this when I have time.

**Comment by aubergine10** [comment]
Good catch!!

**Comment by kianzarrin** [comment]
@Elesbaan70  is this true? if so have you already fixed this?

If this is fixed or is a duplicate please close the issue.

**Comment by Elesbaan70** [comment]
Yes, this is resolved by my PR. I'll mark the PR as resolving this when merged.
